### PR TITLE
fix golint errors in pkg/kubelet/*

### PR DESCRIPTION
**Pull request by tianshapjq**
**What this PR does / why we need it**:
fix golint errors in pkg/kubelet, I will keep submitting commits for the rest packages.
ref #68026 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
@dims 

**Release note**:

```release-note
none
```

### CODE REVIEW

File: pkg/kubelet/sysctl/runtime.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ import (
 )
 
 const (
+	// UnsupportedReason represents unsupported sysctl
```
Comment (tallclair) [inline pkg/kubelet/sysctl/runtime.go]: Change to:

> UnsupportedReason is the reason set in the failed pod status when a sysctl is unsupported

File: pkg/kubelet/sysctl/runtime.go [hunk 2]
```diff
@@ -35,15 +36,16 @@ const (
 // TODO: The admission logic in this file is runtime-dependent. It should be
 // changed to be generic and CRI-compatible.
 
-type runtimeAdmitHandler struct {
+// RuntimeAdmitHandler handles admit for runtime
+type RuntimeAdmitHandler struct {
```
Comment (tallclair) [inline pkg/kubelet/sysctl/runtime.go]: Why did this change to being exported?

File: pkg/kubelet/sysctl/whitelist.go [hunk 3]
```diff
@@ -26,23 +26,23 @@ import (
 )
 
 const (
-	AnnotationInvalidReason = "InvalidSysctlAnnotation"
-	ForbiddenReason         = "SysctlForbidden"
+	// ForbiddenReason represents reason to forbidden sysctl
```
Comment (tallclair) [inline pkg/kubelet/sysctl/whitelist.go]: Change to:

> ForbiddenReason is the reason set in the failed pod status when a sysctl is explicitly forbidden

File: pkg/kubelet/sysctl/whitelist.go [hunk 4]
```diff
@@ -26,23 +26,23 @@ import (
...
+	ForbiddenReason = "SysctlForbidden"
 )
 
-// patternWhitelist takes a list of sysctls or sysctl patterns (ending in *) and
+// PatternWhitelist takes a list of sysctls or sysctl patterns (ending in *) and
 // checks validity via a sysctl and prefix map, rejecting those which are not known
 // to be namespaced.
-type patternWhitelist struct {
+type PatternWhitelist struct {
```
Comment (tallclair) [inline pkg/kubelet/sysctl/whitelist.go]: Why is this being exported?

File: pkg/kubelet/types/constants.go [hunk 5]
```diff
@@ -17,16 +17,22 @@ limitations under the License.
 package types
 
 const (
-	// system default DNS resolver configuration
+	// ResolvConfDefault represents system default DNS resolver configuration
 	ResolvConfDefault = "/etc/resolv.conf"
 
-	// different container runtimes
+	// DockerContainerRuntime represent docker runtime
```
Comment (tallclair) [inline pkg/kubelet/types/constants.go]: Change to:

> DockerContainerRuntime is the name of the built-in docker shim runtime.

File: pkg/kubelet/types/constants.go [hunk 6]
```diff
@@ -17,16 +17,22 @@ limitations under the License.
...
 	DockerContainerRuntime = "docker"
+	// RemoteContainerRuntime represent remote runtime
```
Comment (tallclair) [inline pkg/kubelet/types/constants.go]: Change to:

> RemoteContainerRuntime is the name of the runtime using an out-of-process CRI server.

File: pkg/kubelet/types/constants.go [hunk 7]
```diff
@@ -17,16 +17,22 @@ limitations under the License.
...
 	RemoteContainerRuntime = "remote"
 
-	// User visible keys for managing node allocatable enforcement on the node.
+	// Following is user visible keys for managing node allocatable enforcement on the node.
+
+	// NodeAllocatableEnforcementKey represents key for pods allocatable
```
Comment (tallclair) [inline pkg/kubelet/types/constants.go]: Use more descriptive comments. See https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/node-allocatable.md#phase-2---enforce-allocatable-on-pods

File: pkg/kubelet/types/labels.go [hunk 8]
```diff
@@ -17,25 +17,34 @@ limitations under the License.
 package types
 
 const (
-	KubernetesPodNameLabel       = "io.kubernetes.pod.name"
-	KubernetesPodNamespaceLabel  = "io.kubernetes.pod.namespace"
-	KubernetesPodUIDLabel        = "io.kubernetes.pod.uid"
+	// KubernetesPodNameLabel is the label of Kubernetes pod name
```
Comment (tallclair) [inline pkg/kubelet/types/labels.go]: Specifically, these are labels attached to the container by the container runtime.

File: pkg/kubelet/types/pod_update.go [hunk 9]
```diff
@@ -20,52 +20,55 @@ import (
...
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	kubeapi "k8s.io/kubernetes/pkg/apis/core"
 	"k8s.io/kubernetes/pkg/apis/scheduling"
 	"k8s.io/kubernetes/pkg/features"
 )
 
 const (
-	ConfigSourceAnnotationKey    = "kubernetes.io/config.source"
-	ConfigMirrorAnnotationKey    = v1.MirrorPodAnnotationKey
+	// ConfigSourceAnnotationKey is the key for config source annotation
```
Comment (tallclair) [inline pkg/kubelet/types/pod_update.go]: What are these annotations used for? Please be more descriptive.

File: pkg/kubelet/util/pluginwatcher/example_plugin.go [hunk 10]
```diff
@@ -33,8 +33,8 @@ import (
 	v1beta2 "k8s.io/kubernetes/pkg/kubelet/util/pluginwatcher/example_plugin_apis/v1beta2"
 )
 
-// examplePlugin is a sample plugin to work with plugin watcher
-type examplePlugin struct {
+// ExamplePlugin is a sample plugin to work with plugin watcher
+type ExamplePlugin struct {
```
Comment (tallclair) [inline pkg/kubelet/util/pluginwatcher/example_plugin.go]: Why is this being exported?

File: pkg/kubelet/util/pluginwatcher/example_handler.go [hunk 11]
```diff
@@ -30,7 +30,8 @@ import (
 	v1beta2 "k8s.io/kubernetes/pkg/kubelet/util/pluginwatcher/example_plugin_apis/v1beta2"
 )
 
-type exampleHandler struct {
+// ExampleHandler fakes an example of plugin handler for testing
+type ExampleHandler struct {
```
Comment (tallclair) [inline pkg/kubelet/util/pluginwatcher/example_handler.go]: Why is this being exported?

File: pkg/kubelet/sysctl/whitelist.go (hunk 4 above)
Comment (tianshapjq) [reply pkg/kubelet/sysctl/whitelist.go]: how about keeping this unexported and just take other changes? but that we have to keep pkg/kubelet/sysctl in the hack/.golint_failures.

File: pkg/kubelet/types/constants.go (hunk 7 above)
Comment (tianshapjq) [reply pkg/kubelet/types/constants.go]: thanks for your review. I will have more descriptions.

File: pkg/kubelet/sysctl/whitelist.go (hunk 4 above)
Comment (tallclair) [reply pkg/kubelet/sysctl/whitelist.go]: The better fix then is to change NewWhitelist to return a `lifecycle.PodAdmitHandler`

**Pull request closed without merging**